File: benchmarks/closure/closure.lua

local t0 = os.clock()
local N = tonumber(arg[1])

function adder(n)
  function add(x)
      return n + x
  end
  return add
end

local sum = 0
while N > 0 do
  sum = adder(N)(sum)
  N = N - 1
end

print("Sum: " .. sum)

local t1 = os.clock()

print("Execution took (ms) " .. ((t1 - t0) * 1000))

File: benchmarks/closure/closure.py
import sys
import time

N = int(sys.argv[1])

t0 = time.perf_counter()


sum_ = 0
while N > 0:
  sum_ = (lambda x: x + N)(sum_)
  N -= 1

print("Sum: " + str(sum_))

t1 = time.perf_counter()

print("Execution took (ms) " + str((t1 - t0) * 1000))